helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -423,6 +423,13 @@ public boolean update(String path, DataUpdater<T> updater, int options) {
    * sync update
    */
   public AccessResult doUpdate(String path, DataUpdater<T> updater, int options) {
+    return doUpdate(path, updater, options, ZkClient.TTL_NOT_SET);
+  }
+
+  /**
+   * sync update with ttl
+   */
+  public AccessResult doUpdate(String path, DataUpdater<T> updater, int options, long ttl) {
```

Comment:
We should clarify the behavior of calling update with a TTL on a node that already exists. Since ttl is set only on create, I'm assuming calling update with TTL of Y on a node that was created with TTL of X will not change the TTL value from X --> Y ? Maybe we should clarify this behavior in the javadoc

Reply:
right, will clarify in javadoc.